Plugins/GitUIPluginInterfaces/CredentialsManager.cs

```diff
@@ -36,18 +37,23 @@ public void Save()
 
             Credentials.Clear();
 
-            foreach (var networkCredentials in credentials.Where(c => c.Value is not null))
+            foreach (var networkCredentials in credentials)
             {
+                if (networkCredentials.Value is null)
+                {
+                    continue;
+                }
```

Comment:
Nullability doesn't flow well through linq. Make this explicit here.

Reply:
This way is faster too :)

Reply:
It can make a difference on a hot path, especially on enumeration and especially if the value being foreached has a struct enumerator. I doubt we have many credendials here though.